File: aqua-numeric-digits.js
/* Preserve the Finance/Map baseline while accepting Persian/Arabic digits globally. */
(()=>{
  const previous=window.app;if(typeof previous!=='function')return;
  const digits={'۰':'0','۱':'1','۲':'2','۳':'3','۴':'4','۵':'5','۶':'6','۷':'7','۸':'8','۹':'9','٠':'0','١':'1','٢':'2','٣':'3','٤':'4','٥':'5','٦':'6','٧':'7','٨':'8','٩':'9'};
  window.app=function(){
    const state=previous();
    state.num=function(v){return Number(String(v??0).replace(/[۰-۹٠-٩]/g,n=>digits[n]).replace(/[٬،,\s]/g,''))||0};
    return state;
  };
})();

File: aqua_numeric_digits.py
"""Persian/Arabic digit normalization without replacing the finance/map base UI."""
from flask import request

import app_v3


@app_v3.app.get("/aqua-numeric-digits.js")
def aqua_numeric_digits_js():
    return app_v3.send_from_directory(".", "aqua-numeric-digits.js", mimetype="application/javascript", max_age=0)


@app_v3.app.after_request
def inject_aqua_numeric_digits(response):
    if request.path in {"/", "/index.html"} and response.mimetype == "text/html":
        response.direct_passthrough = False
        body = response.get_data(as_text=True)
        if "/aqua-numeric-digits.js" not in body:
            body = body.replace("</body>", '<script src="/aqua-numeric-digits.js?v=20260914-1"></script></body>', 1)
            response.set_data(body)
            response.headers["Content-Length"] = str(len(response.get_data()))
    return response
